### ISSUE TITLE
RequireJS : contexts[contextId].refresh & waypoint.queueTrigger is not a function

### ISSUE THREAD

**Issue by spAnser** [imakewebthings/waypoints#535]
Running into some errors after switching my code over to RequireJS

This triggers most of the time on page loading.
```
noframework.waypoints.js:441 Uncaught TypeError: contexts[contextId].refresh is not a function(…)
```

This only triggers when a waypoint appears on screen.
```
noframework.waypoints.js:289 Uncaught TypeError: waypoint.queueTrigger is not a function(…)
```

**Comment by spAnser** [comment]
I've narrowed down the issue to the `for...in` loops like: [#L101](https://github.com/imakewebthings/waypoints/blob/master/lib/noframework.waypoints.js#L101)

If I change all for loops like

```
for (var waypointKey in allWaypoints) {
}
```

into

```
Object.keys(allWaypoints).forEach(function(waypointKey) {
})
```

It fixes all my errors when requiring the noframework script with requirejs

Might be worth noting that Object.keys().foreach is not ideal for speed.
StackOverflow: [Problem I ran into actually with solution.](http://stackoverflow.com/questions/25052758/for-in-vs-object-key-foreach-without-inherited-properties)

Where this would be the fastest. Also the object keys above breaks some for loops that call `this`.

```
for (var i = 0, keys = Object.keys(allWaypoints); i < keys.length; i++) {
    var waypointKey = keys[i];
}
```
